pytest/lib/bridge.py

```diff
@@ -223,99 +344,163 @@ def __init__(self, config):
                         service),
                     'out.log')).touch()
 
-        assert subprocess.check_output(
-            ['node', 'write_config.js'], cwd=self.bridge_dir) == b''
-
-    def _git_clone_install(self):
-        print('No rainbow-bridge repo found, cloning...')
+        logger.info('Running write_config.js...')
+        assert_success(subprocess.check_output(['node', 'write_config.js'], cwd=self.bridge_dir))
+        logger.info('Setting up users\' addresses...')
+        for user in [bridge_master_account, alice, bob, carol]:
+            user.eth_address = self.get_eth_address(user)
+            user.near_signer_key = Key(user.near_account_name, user.near_pk, user.near_sk)
+            self.create_account(user, node)
+        logger.info('Bridge is started')
+
+    @retry(wait_exponential_multiplier=1.2, wait_exponential_max=10000)
+    def create_account(self, user, node):
+        status = node.get_status(check_storage=False)
+        h = base58.b58decode(status['sync_info']['latest_block_hash'].encode('utf8'))
+        nonce = node.get_nonce_for_pk(node.signer_key.account_id, node.signer_key.pk)
+        create_account_tx = sign_create_account_with_full_access_key_and_balance_tx(
+                node.signer_key,
+                user.near_account_name,
+                user.near_signer_key,
+                billion_tokens // 10,
+                nonce + 1,
+                h)
+        res = node.send_tx_and_wait(create_account_tx, timeout=15)
+        assert_success(res['result']['status']['SuccessValue'])
```

Comment:
Is it the following scenario possible?

- You create a tx and send it.
- After 15 seconds it hasn't passed because of some glitch on the network
- The tx is still around and it eventually succeed
- Since it initially failed, the tx is re-issued, and it will fail now because the account was already created in the previous tx.

The way to address this, is checking first if the account exist before trying to issue the tx. Anyway, 15 seconds is probably enough time, such that the described scenario is very unlikely

Reply:
It might be possible. However, after bunch of executions at Nayduck I haven't seen this scenario happened.
Anyway, this is a good catch, and I'm going to replace it with
```
        res = node.send_tx(create_account_tx)
        assert 'result' in res
```

Reply:
Hm, getting nonce works badly after this fix:
`nonce = node.get_nonce_for_pk(tx.sender.near_account_name, tx.sender.near_signer_key.pk)` returns None. I'm going to investigate.

Reply:
Let me try this one:
```
        res = node.send_tx(create_account_tx)['result']
        while True:
            try:
                created = node.get_tx(res, user.near_account_name)
                assert_success(created['result']['status']['SuccessValue'])
                break
            except:
                time.sleep(1)
```

Reply:
Seems that last one works well. I'm surprised it's not enough to make sure that create account tx is just accepted by network.